src/main/java/com/hubspot/jinjava/util/DeferredValueUtils.java

```diff
@@ -93,7 +95,7 @@ public static void deferVariables(String[] varTokens, Map<String, Object> contex
         eagerToken.getCurrentMacroFunction() == null ||
         eagerToken
           .getCurrentMacroFunction()
-          .equals(context.getMacroStack().peek().orElse(null))
+          .equals(context.getParent().getMacroStack().peek().orElse(null))
```

Comment:
What I'm trying to do here is find the context which the macro function is being called from and not defer certain values on that context level and higher (because within the macro function it is it's own context scope with variable names from its arguments which are isolated to it)

Calls the parent's macro stack because for the eager execution `executeInChildContext` logic, we enter a non-stacking scope which ends up having the parent context think that it has the macro function on the stack, even though it is its child context that the macro function started from.